hub/core/meta/encode/byte_positions.py

```diff
@@ -84,3 +87,18 @@ def _derive_value(self, row: np.ndarray, row_index: int, local_sample_index: int
         start_byte = row_start_byte + (local_sample_index - index_bias) * row_num_bytes
         end_byte = start_byte + row_num_bytes
         return int(start_byte), int(end_byte)
+
+    @classmethod
+    def frombuffer(cls, buffer: bytes):
```

Comment:
we already have write_byte_positions similar to tobytes and something similar to frombuffer in deserialize chunk. maybe we can combine these